checker/src/body_visitor.rs

```diff
@@ -1821,6 +1821,15 @@ impl<'analysis, 'compilation, 'tcx, E> BodyVisitor<'analysis, 'compilation, 'tcx
         // be saddled with removing it. This case corresponds to no_children being true.
         if val_type == ExpressionType::NonPrimitive {
             // First look at paths that are rooted in rpath.
+            // Remove an explicit deref from source_path to make it a valid root.
```

Comment:
This may also happen in other methods. Do we need to check all the sites where a path is used as a root?

Reply:
I've checked all the sites and this is only place where it can happen.